wgpu/Cargo.toml

```diff
@@ -100,7 +100,7 @@ optional = true
 [dependencies.wgt]
 workspace = true
 
-[target.'cfg(any(not(target_arch = "wasm32"), target_os = "emscripten"))'.dependencies.hal]
```

Comment:
What's up with this?

Reply:
Oh I see the problem. This change makes wgpu-hal a dependency even when targetting WebGPU. We need to follow the dependency model of wgpu-core here, where it's optional on wasm, only enabled if the webgl/emscripten.

Reply:
Done, I think?

It'd be nice if there was big-picture documentation explaining the intended crate relationships and API/abstraction layering. That would have helped me write even the beginnings of this PR, since there are so many layers that I had to adjust to match each other and test on multiple feature sets.

Reply:
Agreed.